### ISSUE TITLE
lz4 writer does not close properly

### ISSUE THREAD

**Issue by rongcuid** [PSeitz/lz4_flex#94]
When I use `FrameEncoder`, the last part of the file is not actually written properly:

```rust
let output = Box::new(lz4_flex::frame::FrameEncoder::new(File::create(output)?));
// many write calls
output.flush()?;
```

Using `lz4cat` on resulting file gives `Error 68: unfinished stream`.

I need this boxing because I am generalizing multiple output backends with `Box<dyn Write>`. I am not sure if this is causing the issue here, but `flate2` with its gzip decoder closes files properly.

**Comment by rongcuid** [comment]
As a note, since I am using `Box<dyn Writer>`, I cannot actually call `.finish()`. In my opinion, this should really be called on `drop()`.

**Comment by PSeitz** [comment]
Thanks for the suggestion, I added `auto_finish` for `FrameEncoder` on the main branch. Can you test if that works for you?

**Comment by rongcuid** [comment]
I think it works.

**Comment by PSeitz** [comment]
Okay cool, I'll try to make a release next week

**Comment by rongcuid** [comment]
I'd just like to check on the status, thanks

**Comment by PSeitz** [comment]
Related issue https://github.com/ouch-org/ouch/pull/163#discussion_r1108965151

**Comment by PSeitz** [comment]
Released with `0.11`
